File: 1713-minimum-operations-to-make-a-subsequence/1713-minimum-operations-to-make-a-subsequence.py
class Solution:
    def minOperations(self, target: List[int], arr: List[int]) -> int:
        m = {x : i for i, x in enumerate(target)}
        dp = []
        for x in arr:
            if x not in m: continue
            if not dp or m[x] > dp[-1]: dp.append(m[x])
            else: dp[bisect_left(dp, m[x])] = m[x]
        return len(target) - len(dp)

File: 1713-minimum-operations-to-make-a-subsequence/1713-minimum-operations-to-make-a-subsequence.cpp
class Solution {
public:
    int minOperations(vector<int>& target, vector<int>& arr) {
        unordered_map<int, int> m;
        int n = target.size();
        
        for(int i = 0; i < n; ++i) m[target[i]] = i;
        vector<int> dp;
        for(int x : arr) {
            auto idx = m.find(x);
            if(idx == m.end()) continue;
            int id = idx->second;
            if(dp.empty() || id > dp.back()) dp.push_back(id);
            else *lower_bound(dp.begin(), dp.end(), id) = id;
        }
        
        return n - dp.size();
    }
};